frame/balances/src/lib.rs

```diff
@@ -608,1602 +466,615 @@ pub mod pallet {
-		Self::try_mutate_account(who, |a, _| -> Result<R, DispatchError> { Ok(f(a)) })
-	}
+		/// Transfer the entire transferable balance from the caller account.
+		///
+		/// NOTE: This function only attempts to transfer _transferable_ balances. This means that
+		/// any locked, reserved, or existential deposits (when `keep_alive` is `true`), will not be
+		/// transferred by this function. To ensure that this function results in a killed account,
+		/// you might need to prepare the account by removing any reference counters, storage
+		/// deposits, etc...
+		///
+		/// The dispatch origin of this call must be Signed.
+		///
+		/// - `dest`: The recipient of the transfer.
+		/// - `keep_alive`: A boolean to determine if the `transfer_all` operation should send all
+		///   of the funds the account has, causing the sender account to be killed (false), or
+		///   transfer everything except at least the existential deposit, which will guarantee to
+		///   keep the sender account alive (true). # <weight>
+		/// - O(1). Just like transfer, but reading the user's transferable balance first.
+		///   #</weight>
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::transfer_all())]
+		pub fn transfer_all(
+			origin: OriginFor<T>,
+			dest: AccountIdLookupOf<T>,
+			keep_alive: bool,
+		) -> DispatchResult {
+			let transactor = ensure_signed(origin)?;
+			let keep_alive = if keep_alive { Keep } else { CanKill };
+			let reducible_balance =
+				<Self as fungible::Inspect<_>>::reducible_balance(&transactor, keep_alive, false);
+			let dest = T::Lookup::lookup(dest)?;
+			<Self as fungible::Mutate<_>>::transfer(
+				&transactor,
+				&dest,
+				reducible_balance,
+				keep_alive,
+			)?;
+			Ok(())
+		}
 
-	/// Mutate an account to some new value, or delete it entirely with `None`. Will enforce
-	/// `ExistentialDeposit` law, annulling the account as needed. This will do nothing if the
-	/// result of `f` is an `Err`.
-	///
-	/// NOTE: Doesn't do any preparatory work for creating a new account, so should only be used
-	/// when it is known that the account already exists.
-	///
-	/// NOTE: LOW-LEVEL: This will not attempt to maintain total issuance. It is expected that
-	/// the caller will do this.
-	fn try_mutate_account<R, E: From<DispatchError>>(
-		who: &T::AccountId,
-		f: impl FnOnce(&mut AccountData<T::Balance>, bool) -> Result<R, E>,
-	) -> Result<R, E> {
-		Self::try_mutate_account_with_dust(who, f).map(|(result, dust_cleaner)| {
-			drop(dust_cleaner);
-			result
-		})
-	}
+		/// Unreserve some balance from a user by force.
+		///
+		/// Can only be called by ROOT.
+		#[pallet::call_index(5)]
+		#[pallet::weight(T::WeightInfo::force_unreserve())]
+		pub fn force_unreserve(
+			origin: OriginFor<T>,
+			who: AccountIdLookupOf<T>,
+			amount: T::Balance,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			let who = T::Lookup::lookup(who)?;
+			let _leftover = <Self as ReservableCurrency<_>>::unreserve(&who, amount);
```

Comment:
Could not this lead to some inconsistency? Since no reason given,  no attempt to update the `Holds`.
Still need to understand the migration process better.

Reply:
Yes - this is really for pre-existing anonymous reserves under the old API. It will be removed once all pallets have been transitioned to the new API.